smoke-tests/src/test/groovy/io/opentelemetry/smoketest/SmokeTest.groovy

```diff
@@ -243,4 +219,59 @@ abstract class SmokeTest extends Specification {
     }
     return encoding
   }
+
+  static class Backend {
+    private static final INSTANCE = new Backend()
+
+    private final Network network = Network.newNetwork()
+    private GenericContainer backend
+    private GenericContainer collector
+
+    boolean started = false
+
+    static Backend getInstance() {
+      return INSTANCE
+    }
+
+    def setup() {
+      // we start backend & collector once for all tests
+      if (started) {
+        return
+      }
+      started = true
```

Comment:
Do smoke tests run concurrently? If so, consider using an `AtomicBoolean` instead (and making the method synchronized)

Reply:
My understanding is that tests run serially within a process, there can be multiple process, but they don't share anything. It is possible to make junit run tests in parallel with multiple threads but we don't use it.